File: project3-template/src/main/java/Mason.java
import  java.util.ArrayList;
import  java.util.List;

// Define function called intersect that takes in 2 lists
// Returns a list
public class Mason {
    public static <T> List<T> intersect(List<T> list1, List<T> list2) {
        // if list1 is null, return empty list
        if (list1 == null) {
            return new ArrayList<>();
        }
        // (memq (car lizt1) list2)
        //     (cons (car lizt1) (intersect (cdr lizt1) lizt2))

        else // return intersect passed cdr of list
        {
            return null; // stub
        }
    }
}


// 

File: project3-template/src/main/clojure/Mason.clj
(ns Mason)

(defn intersect [lizt1, lizt2]
    (cond
        (empty? lizt1) '()
        (some #(= (first lizt1) %) lizt2)
            (cons (first lizt1) (intersect (rest lizt1) lizt2))
        :else (
            intersect (rest lizt1) lizt2)
    )
)